newIDE/app/src/ResourcesList/ResourceSelector.js

```diff
@@ -124,42 +130,44 @@ export default class ResourceSelector extends React.Component<Props, State> {
 
-    const { project, onChooseResource } = this.props;
-    onChooseResource({
-      initialSourceName: source.name,
-      multiSelection: false,
-      resourceKind: this.props.resourceKind,
-    })
-      .then(resources => {
-        if (!resources.length) return;
-        const resource = resources[0];
-        applyResourceDefaults(project, resource);
+      const { project, resourceManagementProps } = this.props;
+      const resources = await resourceManagementProps.onChooseResource({
+        initialSourceName: source.name,
+        multiSelection: false,
+        resourceKind: this.props.resourceKind,
+      });
 
-        // addResource will check if a resource with the same name exists, and if it is
-        // the case, no new resource will be added.
-        project.getResourcesManager().addResource(resource);
+      if (!resources.length) return;
+      const resource = resources[0];
+      applyResourceDefaults(project, resource);
 
-        this._loadFrom(project.getResourcesManager());
-        const resourceName: string = resource.getName();
-        this._onChangeResourceName(resourceName);
+      // addResource will check if a resource with the same name exists, and if it is
+      // the case, no new resource will be added.
+      project.getResourcesManager().addResource(resource);
 
-        // Imperatively set the value of the autocomplete, as it can be (on Windows for example),
-        // still focused. This means that when it's then getting blurred, the value we
-        // set for the resource name would get erased by the one that was getting entered.
-        if (this._autoComplete)
-          this._autoComplete.forceInputValueTo(resourceName);
+      this._loadFrom(project.getResourcesManager());
+      const resourceName: string = resource.getName();
+      this._onChangeResourceName(resourceName);
 
-        // Important, we are responsible for deleting the resources that were given to us.
-        // Otherwise we have a memory leak, as calling addResource is making a copy of the resource.
-        resources.forEach(resource => resource.delete());
-      })
-      .catch(err => {
-        // Should never happen, errors should be shown in the interface.
-        console.error('Unable to choose a resource', err);
-      });
+      // Imperatively set the value of the autocomplete, as it can be (on Windows for example),
+      // still focused. This means that when it's then getting blurred, the value we
+      // set for the resource name would get erased by the one that was getting entered.
+      if (this._autoComplete)
+        this._autoComplete.forceInputValueTo(resourceName);
+
+      // Important, we are responsible for deleting the resources that were given to us.
+      // Otherwise we have a memory leak, as calling addResource is making a copy of the resource.
+      resources.forEach(resource => resource.delete());
+
+      await resourceManagementProps.onFetchNewlyAddedResources();
```

Comment:
why is this now needed?

Reply:
This is for when you select a resource - there is a chance that you use a resource from the "resource store" (this is something new for the desktop app). In this case, the resource is a URL. For the web-app, this is already good. For the desktop-app, which now also has this "resource store", the resource must be "fetched". 
This means it will:
- Do nothing for the web-app (see BrowserResourceFetcher).
- Will download the file if it's the desktop app (and you're a local project) (see LocalResourceFetcher).